Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -675,6 +701,69 @@ void MDNorm::createNormalizationWS(
   }
 }
 
+/**
+ * Validates the TemporaryDataWorkspace has the same binning
+ * as the input binning parameters
+ * @param parameters :: map of binning parameters
+ * @param tempDataWS :: the workspace weare using to aggregate from
+ * @return :: bool - true means the binning is correct to aggreagete using
+ * tempDataWS
+ */
+bool MDNorm::isValidBinningForTemporaryDataWorkspace(
+    const std::map<std::string, std::string> &parameters,
+    const Mantid::API::IMDHistoWorkspace_sptr tempDataWS) {
+
+  std::string numBinsStr = parameters.at("OutputBins");
+  std::string extentsStr = parameters.at("OutputExtents");
+  std::string tmp;
```

Comment:
Check dimension names as well. I know you only do Q_sample, but someone might use a different workspace.